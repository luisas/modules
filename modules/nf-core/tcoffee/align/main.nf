process TCOFFEE_ALIGN {
    tag "$meta.id"
    label 'process_medium'

    input:
    tuple val(meta) ,  path(fasta)
    tuple val(meta2),  path(tree)
    tuple val(meta3),  path(template), path(accessory_informations)

    output:
    tuple val (meta), path ("*.aln"), emit: msa
    path "versions.yml" , emit: versions

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """    
    t_coffee -seq ${fasta} \
        $args \
        -outfile ${prefix}.aln


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        t_coffee: \$( t_coffee -version | sed 's/.*(Version_\\(.*\\)).*/\\1/' )
    END_VERSIONS
    """
    
    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.aln

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        t_coffee: \$( t_coffee -version | sed 's/.*(Version_\\(.*\\)).*/\\1/' )
    END_VERSIONS
    """
}
