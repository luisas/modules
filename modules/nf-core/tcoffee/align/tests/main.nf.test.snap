{
    "align_sequence - uncompressed - seatoxin": {
        "content": [
            [
                "versions.yml:md5,95c81894b17f1e7cd917df513c27458e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T08:01:20.749130634"
    },
    "align_with_structure - uncompressed - seatoxin": {
        "content": [
            [
                "versions.yml:md5,95c81894b17f1e7cd917df513c27458e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T08:01:53.240805233"
    },
    "align_sequence - compressed - seatoxin": {
        "content": [
            [
                "versions.yml:md5,95c81894b17f1e7cd917df513c27458e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T08:01:05.500957902"
    },
    "align_with_guide_tree - uncompressed - seatoxin": {
        "content": [
            [
                "versions.yml:md5,95c81894b17f1e7cd917df513c27458e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T08:01:36.781063137"
    },
    "align_with_lib - seatoxin": {
        "content": [
            [
                "versions.yml:md5,95c81894b17f1e7cd917df513c27458e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T08:02:04.884176473"
    },
    "align_sequence - uncompressed - seatoxin - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.aln:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,95c81894b17f1e7cd917df513c27458e"
                ],
                "alignment": [
                    [
                        {
                            "id": "test"
                        },
                        "test.aln:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "lib": [
                    
                ],
                "versions": [
                    "versions.yml:md5,95c81894b17f1e7cd917df513c27458e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T08:02:17.843814218"
    }
}